app/models/host/managed.rb

```diff
@@ -28,7 +28,7 @@ class Host::Managed < Host::Base
   has_many :all_reports, :foreign_key => :host_id
 
   belongs_to :image
-  has_many :host_statuses, -> { where.not(type: nil) }, :class_name => 'HostStatus::Status', :foreign_key => 'host_id', :inverse_of => :host, :dependent => :destroy
+  has_many :host_statuses, :class_name => 'HostStatus::Status', :foreign_key => 'host_id', :inverse_of => :host, :dependent => :destroy
```

Comment:
Wouldn't duplicating the association with negated condition be less error prone? I'm worried that it is just a question of time before someone forgets to filter for non-null statuses when they access them

```suggestion
  has_many :host_statuses, -> { where.not(type: nil) }, :class_name => 'HostStatus::Status', :foreign_key => 'host_id', :inverse_of => :host, :dependent => :destroy
  has_many :host_null_statuses, -> { where(type: nil) }, :class_name => 'HostStatus::Status', :foreign_key => 'host_id', :inverse_of => :host, :dependent => :destroy

```

Reply:
Thanks, this seems to me like the best "silent" solution to this issue. I am going to try it out.